src/main/java/com/faforever/client/remote/ReconnectService.java

```diff
@@ -0,0 +1,63 @@
+package com.faforever.client.remote;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+@Service
+@Slf4j
+public class ReconnectService {
```

Comment:
I am unclear about responsibilities here. I would assume a reconnect service is responsible for invoking the reconnect. But as it seems it is merely a simple util.

Reply:
Renamed it